lib/spack/spack/spec.py

```diff
@@ -2594,6 +2594,16 @@ def _install_status(self):
         except KeyError:
             return None
 
+    def _installed_explicitly(self):
```

Comment:
Maybe this function can be factored with the one above (or we can just leave this for a future PR).